staging/src/k8s.io/apiserver/pkg/server/deleted_kinds.go

```diff
@@ -41,7 +41,16 @@ type resourceExpirationEvaluator struct {
 	serveRemovedAPIsOneMoreRelease bool
 }
 
-func newResourceExpirationEvaluator(currentVersion apimachineryversion.Info) (*resourceExpirationEvaluator, error) {
+// ResourceExpirationEvaluator indicates whether or not a resource should be served.
+type ResourceExpirationEvaluator interface {
```

Comment:
what's the value of the abstraction via an interface?

Reply:
> what's the value of the abstraction via an interface?

Encapsulation, future mocks, potentially choosing a different reason to suppress in the future.